types/wlr_idle_inhibit_v1.c

```diff
@@ -0,0 +1,188 @@
+	.destroy = idle_inhibit_inhibitor_v1_handle_destroy,
+};
+
+static void wlr_create_inhibitor(struct wl_client *client,
+		struct wl_resource *resource, uint32_t id,
+		struct wl_resource *surface_resource) {
+	struct wlr_surface *surface = wlr_surface_from_resource(surface_resource);
+	struct wlr_idle_inhibit_manager_v1 *manager =
+		wlr_idle_inhibit_manager_v1_from_resource(resource);
+	assert(surface && manager);
+
+	struct wlr_idle_inhibit_inhibitor_v1 *inhibitor =
+		calloc(1, sizeof(struct wlr_idle_inhibit_inhibitor_v1));
+	if (!inhibitor) {
+		wl_client_post_no_memory(client);
+		return;
+	}
+
+	struct wl_resource *wl_resource = wl_resource_create(client,
+		&zwp_idle_inhibitor_v1_interface, 1, id);
+	if (!wl_resource) {
+		wl_client_post_no_memory(client);
+		free(inhibitor);
+		return;
+	}
+
+	inhibitor->resource = wl_resource;
+	inhibitor->surface = surface;
+	wl_signal_init(&inhibitor->events.destroy);
+
+	inhibitor->surface_destroy.notify = idle_inhibit_inhibitor_handle_surface_destroy;
+	wl_signal_add(&surface->events.destroy, &inhibitor->surface_destroy);
+
+
+	wl_resource_set_implementation(wl_resource, &idle_inhibitor_impl,
+		inhibitor, idle_inhibit_inhibitor_destroy);
+
+	wl_list_insert(&manager->inhibitors, &inhibitor->link);
+	wlr_signal_emit_safe(&manager->events.new_inhibitor, inhibitor);
+}
+
+
+static void idle_inhibit_manager_v1_destroy(struct wl_resource *resource) {
+	wl_list_remove(wl_resource_get_link(resource));
+}
+
+static void idle_inhibit_manager_v1_handle_destroy(struct wl_client *client,
+		struct wl_resource *manager_resource) {
+	wl_resource_destroy(manager_resource);
+}
+
+static struct zwp_idle_inhibit_manager_v1_interface idle_inhibit_impl = {
+	.destroy = idle_inhibit_manager_v1_handle_destroy,
+	.create_inhibitor = wlr_create_inhibitor,
+};
+
+static void handle_display_destroy(struct wl_listener *listener, void *data) {
+	struct wlr_idle_inhibit_manager_v1 *idle_inhibit =
+		wl_container_of(listener, idle_inhibit, display_destroy);
+
+	wlr_idle_inhibit_v1_destroy(idle_inhibit);
+}
+
+static void idle_inhibit_bind(struct wl_client *wl_client, void *data,
+		uint32_t version, uint32_t id) {
+	struct wlr_idle_inhibit_manager_v1 *idle_inhibit = data;
+	assert(wl_client && idle_inhibit);
+
+	struct wl_resource *wl_resource  = wl_resource_create(wl_client,
+		&zwp_idle_inhibit_manager_v1_interface, version, id);
+
+	if (!wl_resource) {
+		wl_client_post_no_memory(wl_client);
+		return;
+	}
+
+	wl_list_insert(&idle_inhibit->wl_resources, wl_resource_get_link(wl_resource));
+
+	wl_resource_set_implementation(wl_resource, &idle_inhibit_impl,
+		idle_inhibit, idle_inhibit_manager_v1_destroy);
+	wlr_log(L_DEBUG, "idle_inhibit bound");
+}
+
+void wlr_idle_inhibit_v1_destroy(struct wlr_idle_inhibit_manager_v1 *idle_inhibit) {
+	if (!idle_inhibit) {
+		return;
+	}
+
+	wl_list_remove(&idle_inhibit->display_destroy.link);
+
+	struct wlr_idle_inhibit_inhibitor_v1 *inhibitor;
+	struct wlr_idle_inhibit_inhibitor_v1 *tmp;
```

Comment:
This can be on the same line as the other declaration

Reply:
Unless that's officially part of the coding guidelines I would rather not, since I consider it horrible style.

And I would also like to point towards the kernel coding guidelines (which are referenced as base in the wlroots guidelines) and quote:

> It's also important to comment data, whether they are basic types or derived
> types.  To this end, use just one data declaration per line (no commas for
> multiple data declarations).  This leaves you room for a small comment on each
> item, explaining its use.